sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -445,7 +446,8 @@ case class AlterTableChangeColumnCommand(
   // name(by resolver) and dataType.
   private def columnEqual(
       field: StructField, other: StructField, resolver: Resolver): Boolean = {
-    resolver(field.name, other.name) && field.dataType == other.dataType
+    resolver(field.name, other.name) &&
+      DataType.equalsIgnoreCompatibleCollationAndNullability(field.dataType, other.dataType)
```

Comment:
The comment of the command in line 359 is not outdated

Reply:
Fixed, please check.